src/code.cloudfoundry.org/nats-v2-migrate/server/main.go

```diff
@@ -0,0 +1,162 @@
+package main
+
+import (
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"os/exec"
+
+	"code.cloudfoundry.org/lager"
+	"code.cloudfoundry.org/lager/lagerflags"
+	"code.cloudfoundry.org/nats-v2-migrate/config"
+)
+
+var (
+	gCfg config.Config
+)
+
+func main() {
+	configFilePath := flag.String("config-file", "", "path to config file")
+	flag.Parse()
+
+	cfg, err := config.NewConfig(*configFilePath)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Error reading config file: %v\n", err)
+		os.Exit(1)
+	}
+
+	gCfg = cfg
+
+	logger, _ := lagerflags.NewFromConfig("nats-migrate-server", lagerflags.LagerConfig{LogLevel: lagerflags.INFO, TimeFormat: lagerflags.FormatRFC3339})
+	server := NewHttpServer(logger)
+
+	http.HandleFunc("/info", server.Info)
+	http.HandleFunc("/migrate", server.Migrate)
+
+	logger.Info("Server listening for migration...")
+	http.ListenAndServeTLS(fmt.Sprintf(":%d", cfg.NATSMigratePort), cfg.NATSMigrateServerClientCertFile, cfg.NATSMigrateServerClientKeyFile, nil)
+}
+
+type httpServer struct {
+	logger             lager.Logger
+	migrateEndpointHit bool
+}
+
+func NewHttpServer(logger lager.Logger) *httpServer {
+	return &httpServer{
+		logger:             logger,
+		migrateEndpointHit: false,
+	}
+}
+
+func (s *httpServer) Info(w http.ResponseWriter, req *http.Request) {
+	w.Header().Set("Content-Type", "application/json")
+	response := make(map[string]bool)
+
+	response["bootstrap"] = gCfg.Bootstrap
+
+	jsonResponse, err := json.Marshal(response)
+	if err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+		s.logger.Error("Error during marshal", err)
+		w.Write(nil)
+		return
+	}
+
+	w.WriteHeader(http.StatusOK)
+	w.Write(jsonResponse)
+}
+
+func (s *httpServer) Migrate(w http.ResponseWriter, req *http.Request) {
+	// guard against race condition, multiple hits from different
+	// instances
+	if s.migrateEndpointHit == true {
+		w.WriteHeader(http.StatusConflict)
+		w.Write(nil)
+		return
+	}
+
+	s.migrateEndpointHit = true
+
+	err := replaceBPMConfig(gCfg.NATSBPMv2ConfigPath, gCfg.NATSBPMConfigPath)
+	if err != nil {
+		s.logger.Error("Failed to replace bpm config file", err)
+		w.WriteHeader(http.StatusInternalServerError)
+		w.Write(nil)
+		shutdownNATS(s.logger)
+		return
+	}
+
+	w.WriteHeader(http.StatusOK)
```

Comment:
We should move this after restartNATS (:103) If we remove `depends on nats` in monit file restarting nats should not trigger migration server restart